core/state.go

```diff
@@ -414,11 +418,13 @@ func (s *State) updateContractStorages(stateTrie *trie.Trie, diffs map[felt.Felt
 		return err
 	}
 
+	sort.Slice(bufferedTxns, func(i, j int) bool {
```

Comment:
you could create a private struct inside the function to allow you to use the `sort.Slice` function instead of declaring a `BufferedTransactionWithAddress` which is publically accessible.